addons/website_sale/controllers/main.py

```diff
@@ -1551,7 +1551,7 @@ def shop_payment_validate(self, sale_order_id=None, **post):
             order = request.env['sale.order'].sudo().browse(sale_order_id)
             assert order.id == request.session.get('sale_last_order_id')
 
-        tx = order.get_portal_last_transaction()
+        tx = order.get_portal_last_transaction() if order else None
```

Comment:
```suggestion
        tx = order.get_portal_last_transaction() if order else order.env['payment.transaction']
```
Let's keep the variable type consistent.